### PR TITLE
Update whitelisted hosts in Modrinth modpacks

### PR DESCRIPTION
**Pull request by flowln**
Due to CF being CF, [Modrinth will disallow direct CDN links from CF](https://blog.modrinth.com/modpack-changes/) in the modpack specification. This removes such support, while addind `gitlab.com` as a new whitelisted host. This also makes it so that non-whitelisted URLs prompt a warning window instead of failing.

Also fixes a missing `tr()` btw :D

### CODE REVIEW

**Comment by LennyMcLennington** [comment]
Exception descriptions are for the developers to read and understand, not to be translated into the user's language so we don't know what the exception was.

File: launcher/InstanceImportTask.cpp
Comment (LennyMcLennington) [inline launcher/InstanceImportTask.cpp]: Exception descriptions are for the developers to read and understand, not to be translated into the user's language so we don't know what the exception was. The "missing" tr() is by design. This is why nothing in the debug logs is translated either.

**Review by LennyMcLennington — CHANGES_REQUESTED** [review]

File: launcher/InstanceImportTask.cpp
Comment (flowln) [reply launcher/InstanceImportTask.cpp]: This shows in the error dialog shown to the user

Comment (LennyMcLennington) [reply launcher/InstanceImportTask.cpp]: > This shows in the error dialog shown to the user

That makes sense to translate then, maybe it should be accompanied by an untranslated description of the error in qDebug?

Comment (flowln) [reply launcher/InstanceImportTask.cpp]: Sure!

**Review by LennyMcLennington — APPROVED** [review]
I think everything is OK with your changes now but I will suggest a change that you might as well do in this PR while you're at it. You're welcome to accept it or not accept it, just a suggestion.
```diff
From f8b73c201fec7c6685d72cc60d342ce6600a756f Mon Sep 17 00:00:00 2001
From: Lenny McLennington <lennymclennington@protonmail.com>
Date: Sat, 28 May 2022 19:54:00 +0100
Subject: [PATCH] clean up validateDownloadUrl

---
 .../modrinth/ModrinthPackManifest.cpp         | 20 +++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/launcher/modplatform/modrinth/ModrinthPackManifest.cpp b/launcher/modplatform/modrinth/ModrinthPackManifest.cpp
index 8b379480..33116231 100644
--- a/launcher/modplatform/modrinth/ModrinthPackManifest.cpp
+++ b/launcher/modplatform/modrinth/ModrinthPackManifest.cpp
@@ -42,6 +42,8 @@
 #include "minecraft/MinecraftInstance.h"
 #include "minecraft/PackProfile.h"
 
+#include <QSet>
+
 static ModrinthAPI api;
 
 namespace Modrinth {
@@ -95,17 +97,15 @@ void loadIndexedVersions(Modpack& pack, QJsonDocument& doc)
 
 auto validateDownloadUrl(QUrl url) -> bool
 {
+    static QSet<QString> domainWhitelist{
+        "cdn.modrinth.com",
+        "github.com",
+        "raw.githubusercontent.com",
+        "gitlab.com"
+    };
+
     auto domain = url.host();
-    if(domain == "cdn.modrinth.com")
-        return true;
-    if(domain == "github.com")
-        return true;
-    if(domain == "raw.githubusercontent.com")
-        return true;
-    if(domain == "gitlab.com")
-        return true;
-
-    return false;
+    return domainWhitelist.contains(domain);
 }
 
 auto loadIndexedVersion(QJsonObject &obj) -> ModpackVersion
-- 
2.36.1
```
Haven't tested this yet so you're welcome to test it yourself or I will tell you when I've tested it, didn't want to delay the approval while waiting to test my suggestion.

**Review by Scrumplex — APPROVED** [review]

File: launcher/modplatform/modrinth/ModrinthPackManifest.cpp
Comment (kthchew) [inline launcher/modplatform/modrinth/ModrinthPackManifest.cpp:109]: Forgive me for my ignorance (I don't know anything about the Modrinth modpack format), but why does the launcher need to check for approved download URLs? If Modrinth allows (or bans) certain domains, doesn't that mean that modpacks downloaded from Modrinth will already only include allowed domains, and thus checking again here is redundant and adds extra maintenance?

Comment (flowln) [reply launcher/modplatform/modrinth/ModrinthPackManifest.cpp:109]: Well, it's in the specification so we implemented it. It will be checked by Modrinth themselves, that's right, but `.mrpack`s need not necessarily come from Modrinth, since it's an open format. So, this is just so that we're aligned with Modrinth on what should be allowed.

As of my latest commits, there'll simply be a warning if there's an URL that is not whitelisted, instead of failing, so it prevents things like malicious URLs being thrown at the user without their knowledge, when the pack comes from a third-party.

(Also, maybe you should get into the Discord, we discuss this kinds of things in there x) )

Comment (kthchew) [reply launcher/modplatform/modrinth/ModrinthPackManifest.cpp:109]: Ah, that makes sense.

> (Also, maybe you should get into the Discord, we discuss this kinds of things in there x) )

Ah, sorry for the duplicated answer then. Maybe in the future :) (I don't have a Discord account I want to associate with my GH stuff at the moment :p)

Comment (flowln) [reply launcher/modplatform/modrinth/ModrinthPackManifest.cpp:109]: > Maybe in the future :)

I'll wait for it then :)
(you can also use the matrix bridge if you feel like it ;))

**Review by ZekeZDev — APPROVED** [review]

**Pull request merged**
